src/datepicker/hijri/ngb-date-islamic-civil.ts

```diff
@@ -0,0 +1,91 @@
+import {NgbDateHijri} from './ngb-date-hijri';
+import {Injectable} from '@angular/core';
+
+/**
+ * The civil calendar is one type of Hijri calendars used in islamic countries.
+ * Uses a fixed cycle of alternating 29- and 30-day months,
+ * with a leap day added to the last month of 11 out of every 30 years.
+ * http://cldr.unicode.org/development/development-process/design-proposals/islamic-calendar-types
+ */
+@Injectable()
+export class NgbDateIslamicCivil extends NgbDateHijri {
+  private _GREGORIAN_EPOCH = 1721425.5;
+  private _ISLAMIC_EPOCH = 1948439.5;
+
+  public fromGregorian(gdate: Date): NgbDateIslamicCivil {
+    let date = new Date(gdate);
+    let gYear = date.getFullYear(), gMonth = date.getMonth(), gDay = date.getDate();
+
+    let julianDay = this._GREGORIAN_EPOCH - 1 + 365 * (gYear - 1) + Math.floor((gYear - 1) / 4) +
+        -Math.floor((gYear - 1) / 100) + Math.floor((gYear - 1) / 400) +
+        Math.floor(
+            (367 * (gMonth + 1) - 362) / 12 + (gMonth + 1 <= 2 ? 0 : this._isGregorianLeapYear(date) ? -1 : -2) + gDay);
+    julianDay = Math.floor(julianDay) + 0.5;
+
+    let days = julianDay - this._ISLAMIC_EPOCH;
+    let hYear = Math.floor((30 * days + 10646) / 10631.0);
```

Comment:
All the calculations here are based on the equations from ["Calendrical Calculations" By Edward M. Reingold, Nachum Dershowitz](https://www.amazon.com/Calendrical-Calculations-Nachum-Dershowitz-ebook/dp/B00GA22KBE). You can refer directly to equation (1.77) and its calculation for the Islamic calendar in chapter 1 "Calendar Basics", section 1.12 "Cycles of Years".